packages/react-img-viewer/src/ImageViewer.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useRef, useEffect } from 'react';
+import { FullScreenViewer, ImageViewer } from 'iv-viewer';
+import 'iv-viewer/dist/iv-viewer.css';
+import { ImageViewerProps } from './types';
+
+export const NormalViewer: React.FC<ImageViewerProps> = ({
+  img,
+  hiResImg,
+  width,
+  height,
+  zoom,
+  defaultZoom = 100,
+  maxZoom = 500,
+  snapView = true,
+  refreshOnResize = true,
+  zoomOnMouseWheel = true,
+  hasZoomButtons = true,
+  zoomStep,
+  onZoomChange,
+  onLoad,
+}) => {
+  const viewerRef = useRef<any>(null);
+  const containerRef = useRef<HTMLDivElement>(null);
+  const currentZoom = useRef(defaultZoom);
+
+  const _onZoomChange = useRef(onZoomChange);
+  _onZoomChange.current = onZoomChange;
+  const _onLoad = useRef(onLoad);
+  _onLoad.current = onLoad;
+
+  useEffect(() => {
+    const viewerOptions = {
+      zoom: currentZoom.current,
+      maxZoom,
+      snapView,
+      refreshOnResize,
+      zoomOnMouseWheel,
+      hasZoomButtons,
+      zoomStep,
+      listeners: {
+        onInit: (data: any) => {
+          if (_onLoad.current) {
+            _onLoad.current(data);
+          }
+        },
+        onDestroy: () => {},
+        onZoomChange: (data: any) => {
+          if (_onZoomChange.current) {
+            _onZoomChange.current(data.zoomValue);
+          }
+        },
+      },
+    };
+
+    if (containerRef.current) {
+      viewerRef.current = new ImageViewer(containerRef.current, viewerOptions);
+      viewerRef.current.load(img, hiResImg || null);
+    }
+
+    return () => {
+      if (viewerRef.current) {
+        viewerRef.current.destroy();
+      }
+    };
+  }, [
+    img,
+    hiResImg,
+    maxZoom,
+    snapView,
+    refreshOnResize,
+    zoomOnMouseWheel,
+    hasZoomButtons,
+    zoomStep,
+  ]);
+
+  useEffect(() => {
+    if (zoom && zoom !== currentZoom.current) {
+      currentZoom.current = zoom;
+      if (viewerRef.current && typeof viewerRef.current.zoom === 'function') {
+        viewerRef.current.zoom(zoom);
+      }
+    }
+  }, [zoom]);
+
+  return (
+    <div
+      ref={containerRef}
+      style={{
+        width: width || '100%',
+        height: height || '100%',
+      }}
+    ></div>
+  );
+};
+
+export const FullScreenImageViewer: React.FC<ImageViewerProps> = ({
+  img,
+  hiResImg,
+  defaultZoom = 100,
+  maxZoom = 500,
+  snapView = true,
+  refreshOnResize = true,
+  zoomOnMouseWheel = true,
+  hasZoomButtons = true,
+  zoomStep,
+  onLoad,
+}) => {
+  const viewerRef = useRef<any>(null);
+  const currentZoom = useRef(defaultZoom);
+
+  const _onLoad = useRef(onLoad);
+  _onLoad.current = onLoad;
+
+  useEffect(() => {
+    const viewerOptions = {
+      zoom: currentZoom.current,
+      maxZoom,
+      snapView,
+      refreshOnResize,
+      zoomOnMouseWheel,
+      hasZoomButtons,
+      zoomStep,
+      listeners: {
+        onInit: (data: any) => {
+          if (_onLoad.current) {
+            _onLoad.current(data);
+          }
+        },
```

Comment:
Same, we just need to pass instance on onLoad.
```suggestion
        onInit: () => {
          if (_onLoad.current) {
            _onLoad.current(viewerRef.current);
          }
        },
```

Reply:
viewerRef.current is not initialized on start so it give null in return, that's why we are passing the callback data to the _onLoad.current.
sure i will give the type to the 'data' so there won't be any type error.